### PR TITLE
Created homework_2

### PR DESCRIPTION
**Pull request by Doctor-Ledenec**

### CODE REVIEW

File: src/homework_2/Student.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            course_status = true;
+        } else {
+            int oldLength = this.courses.length;
+            String [] tempArray = new String [oldLength];
+
+            for (int i = 0; i < tempArray.length; i++) {
+                tempArray[i] = this.courses[i];
+            }
+
+            int newLength = oldLength +=1;
```
Comment (SofiaVynnytska12) [inline src/homework_2/Student.java:60]: redundant assignment

File: src/homework_2/Student.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    break;
+                }
+
+                this.grades[i] = tempArray[i];
+            }
+
+            grade_status = true;
+        }
+
+        if (course_status == true && grade_status == true) {
```
Comment (SofiaVynnytska12) [inline src/homework_2/Student.java:104]: simplify the expression

File: src/homework_2/Student.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            result += temp;
+        }
+
+        return result / this.grades.length;
+    }
+
+
+    public boolean addCourseGrade(String course, int grade) {
+
+        boolean course_status = false;
```
Comment (SofiaVynnytska12) [inline src/homework_2/Student.java:44]: use proper Java naming conventions

File: src/homework_2/Student.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        boolean grade_status = false;
```
Comment (SofiaVynnytska12) [inline src/homework_2/Student.java:45]: use proper Java naming conventions

File: src/homework_2/Teacher.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        boolean status = true;
+
+        if (this.courses.length == 0) {
+            status = true;
+            this.courses = new String [1];
+            this.courses [0] = course;
+        } else {
+
+            for (String temp : this.courses) {
+                if (temp == course) {
```
Comment (SofiaVynnytska12) [inline src/homework_2/Teacher.java:36]: incorrect String comparison

File: src/homework_2/Teacher.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                    status = false;
+                    break;
+                }
+            }
+
+            if (status == true) {
```
Comment (SofiaVynnytska12) [inline src/homework_2/Teacher.java:42]: simplify

File: src/homework_2/Teacher.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                int oldLength = this.courses.length;
+                String [] tempArray = new String [oldLength];
+
+                for (int i = 0; i < tempArray.length; i++) {
+                    tempArray[i] = this.courses[i];
+                }
+
+                int newLength = oldLength +=1;
```
Comment (SofiaVynnytska12) [inline src/homework_2/Teacher.java:50]: redundant assignment

File: src/homework_2/Teacher.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            int index = 0;
+            for (int i = 0; i < this.courses.length; i++) {
+                if (courses[i] == course) {
+                    index = i;
+                    status = true;
+                    break;
+                }
+            }
+
+            if (status == true) {
```
Comment (SofiaVynnytska12) [inline src/homework_2/Teacher.java:86]: simplify

**Review by SofiaVynnytska12 — COMMENTED** [review]
Please, resolve merge conflicts. General question : why your classes were added to .gitignore ?

**Comment by Doctor-Ledenec** [comment]
I`ve fixed all of your remerks

**Comment by SofiaVynnytska12** [comment]
Everything is correct now, but the same situation with merge conflicts.

**Comment by Doctor-Ledenec** [comment]
Merge conflicts have been solved

**Pull request closed without merging**